mne/viz/topomap.py

```diff
@@ -1578,11 +1572,267 @@ def _prepare_topomap(pos, ax):
+
+    if params['butterfly']:
+        all_times = params['all_times']
+        line = params['line']
+        line.remove()
+        params['line'], = ax_line.plot([all_times[time_idx],
+                                        all_times[time_idx]],
+                                       ax_line.get_ylim(), color='r')
+        items.append(params['line'])
+    params['frame'] = frame
+    return tuple(items) + tuple(cont.collections)
+
+
+def _pause_anim(event, params):
+    """Function for pausing and continuing the animation on mouse click"""
+    params['pause'] ^= True
+
+
+def _key_press(event, params):
+    """Function for handling key presses for the animation."""
+    if event.key == 'left':
+        params['pause'] = True
+        params['frame'] = max(params['frame'] - 1, 0)
+    elif event.key == 'right':
+        params['pause'] = True
+        params['frame'] = min(params['frame'] + 1, len(params['frames']) - 1)
+
+
+def topomap_animation(evoked, ch_type='mag', times=None, frame_rate=None,
+                      butterfly=False, blit=True):
+    """Make animation of evoked data as topomap timeseries. Animation can be
+    paused/resumed with left mouse button. Left and right arrow keys can be
+    used to move backward or forward in time.
+
+    Parameters
+    ----------
+    evoked : instance of Evoked
+        The evoked data.
+    ch_type : str | None
+        Channel type to plot. Accepted data types: 'mag', 'grad', 'eeg'.
+        If None, first available channel type from ['mag', 'grad', 'eeg'] is
+        used. Defaults to None.
+    times : array of floats | None
+        The time points to plot. If None, 10 evenly spaced samples are
+        calculated over the evoked time series. Defaults to None.
+    frame_rate : int | None
+        Frame rate for the animation in Hz. If None, frame rate = sfreq / 10.
+        Defaults to None.
+    butterfly : bool
+        Whether to plot the data as butterfly plot under the topomap.
+        Defaults to False.
+    blit : bool
+        Whether to use blit to optimize drawing. In general, it is recommended
+        to use blit in combination with ``show=True``. If you intend to save
+        the animation it is better to disable blit. For MacOSX blit is always
+        disabled. Defaults to True.
+
+    Returns
+    -------
+    fig : instance of matplotlib figure
+        The figure.
+    anim : instance of matplotlib FuncAnimation
+        Animation of the topomap.
+
+    Notes
+    -----
+    .. versionadded:: 0.12.0
+    """
+    import matplotlib.pyplot as plt
+    from matplotlib import animation
+    if ch_type is None:
+        ch_type = _picks_by_type(evoked.info)[0][0]
+    if ch_type not in ['mag', 'grad', 'eeg']:
+        raise ValueError("Channel type not supported. Supported channel "
+                         "types include 'mag', 'grad' and 'eeg'.")
+    if times is None:
+        times = np.linspace(evoked.times[0], evoked.times[-1], 10)
+    times = np.array(times)
```

Comment:
throw an informative error if times is not between `evoked.times[0]` and `evoked.times[-1]`?